lib/Sema/CSApply.cpp

```diff
@@ -9132,16 +9132,23 @@ Optional<SolutionApplicationTarget> ConstraintSystem::applySolution(
     Solution &solution, SolutionApplicationTarget target) {
   // If any fixes needed to be applied to arrive at this solution, resolve
   // them to specific expressions.
+  unsigned numResolvableFixes = 0;
   if (!solution.Fixes.empty()) {
     if (shouldSuppressDiagnostics())
       return None;
 
     bool diagnosedErrorsViaFixes = applySolutionFixes(solution);
+    bool canApplySolution = true;
+    for (const auto fix : solution.Fixes) {
+      if (!fix->canApplySolution())
+        canApplySolution = false;
+      if (fix->affectsSolutionScore() == SK_Fix && fix->canApplySolution())
```

Comment:
This seems odd to me because anything which sets `SK_Fix` should imply "fatal"...

Reply:
Good catch, this change became unnecessary with my last commit, because we no longer increase `SK_Fix` for anything recoverable. I can remove it now.

Reply:
Ah, but I've been thinking about adding a `RecoverableError` to `FixBehavior`, which is an error from which we can still recover. We still want it to bump `SK_Fix` so there's no difference from `Error` in the solver behavior, but we can still produce a type-checked AST, which helps downstream code. Adding `@Sendable` in an argument position is a case where this would work, because CSApply will just toss in another function conversion expression.

Reply:
It sounds like we need to split SK_Fix into SK_FatalFix and SK_RecoverableFIx or something like that.